### DISCUSSION TITLE
Login History

### DISCUSSION THREAD

**Discussion by dvuke** [cloudfoundry/uaa#3125] [Error / Help] [answered]
I was wondering if UAA had a way to track login history?  I know the users record has a last logon date stored, but is there anything else that actually tracks the history of successful logins?  Or anything to tie into at all when a successful login occurs so we can track it ourselves?  Our clients want to a dashboard widget tracking successful logins.

**Comment by strehle** [comment, marked as answer]
Hi, there is only the last login, but no history. That would be a new feature

**Comment by sydubabu** [comment]
When I am deploying cloudfoundry-identity-uaa-78.0.0 Getting **Invalid login attempt, the request does not meet our security standards and may indicate that the action was not originated by you. Please try again.** How to fix it? Is there any additional configurations I have to do?

**Reply by strehle** [reply]
This is an issue with Csrf which was fixed with https://github.com/cloudfoundry/uaa/pull/3542 . So use UAA 78.1.0 or even newer one and then it should work

**Comment by sydubabu** [comment]
Hello Cloud Foundry UAA Team,
I’m currently working with Cloud Foundry UAA and have encountered a challenge during an upgrade.
**Current Version: v76.23
Upgraded Version: v78.1.0**

**Context:** In the previous version, I used a servlet defined in web.xml with load-on-startup set to 1. This servlet was responsible for reading values from AWS Secrets Manager and replacing placeholders in the UAA configuration file before the application started.
After upgrading to v78.1.0, I noticed that the configuration has moved to an annotation-based setup due to the migration to Spring Boot. I’m now unsure how to perform placeholder replacement in the config file in this new setup.

Could you please advise on:
  - The recommended approach for injecting secrets from AWS Secrets Manager into the UAA configuration in the Spring Boot-based version.
  - Any documentation or examples that demonstrate how to achieve this in the latest version.

**Reply by strehle** [reply]
Ok for now a safe way to upgrade is 77.xx and here use https://github.com/cloudfoundry/uaa/releases/tag/v77.20.7 which is maintained by Broadcom right now. It gives you CVE free UAA version for now.

**Reply by strehle** [reply]
About AWS Secrets Manager, can  you @fhanik or @duanemay, recommend a good approach with spring boot?
